### ISSUE TITLE
avoid_unused_constructor_parameters lint doesn't support redirecting factory constructors

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/linter#858]
The following gives a warning about unused constructor parameters on the `Foo` factory constructor when it shouldn't afaik.

```dart
class Foo {
  factory Foo({String bar}) = _Foo;
}

class _Foo implements Foo {
  final String bar;
  _Foo({this.bar});
}
```

**Comment by a14n** [comment]
What's your linter version ? I fixed something similar in #847

**Comment by jakemac53** [comment]
Ah yep built the latest sdk and its fixed, thanks and sorry for the noise!
